### PR TITLE
feat(magento2): Update Magento to 2.4.6-p4 and it's PHP to 8.2

### PR DESCRIPTION
**Pull request by andytson-inviqa**
Since 2.4.6, elasticsearch 8 has been supported as well (needs `magento/module-elasticsearch-8` composer package added), but Magento Commerce on Cloud uses opensearch instead

### CODE REVIEW

**Review by kierenevans — DISMISSED** [review]

**Comment by andytson-inviqa** [comment]
spotted some improvements I could make

**Comment by andytson-inviqa** [comment]
and 2.4.6-p3 > 2.4.6 in version_compare correct

File: src/magento2/harness.yml [hunk 1]
```diff
@@ -74,7 +74,7 @@ attributes:
       frontend:
         languages: []
         themes: []
-    version: 2.4.2
+    version: 2.4.6-p3
   mysql:
     image: mariadb
     tag: '10.4'
```
Comment (kierenevans) [inline src/magento2/harness.yml]: Should we upgrade this to 10.6 to match "on-prem" for https://experienceleague.adobe.com/docs/commerce-operations/installation-guide/system-requirements.html ?

Comment (andytson-inviqa) [reply src/magento2/harness.yml]: and even switch to opensearch too

**Review by kierenevans — DISMISSED** [review]

**Review by kierenevans — APPROVED** [review]

File: src/magento2/application/overlay/phpcs.xml [hunk 2]
```diff
@@ -59,7 +59,7 @@
 
     <rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes">
         <properties>
-            <property name="newlinesCountBetweenOpenTagAndDeclare" value="2"/>
+            <property name="linesCountBeforeDeclare" value="1"/>
```
Comment (andytson-inviqa) [inline src/magento2/application/overlay/phpcs.xml:62]: as this is an overlay, projects would need to upgrade slevomat/coding-standard or revert in an override

**Pull request merged**